File: android/src/main/java/com/margelo/nitro/nitrologger/HybridNativeConsoleSink.kt
package com.margelo.nitro.nitrologger

import com.facebook.proguard.annotations.DoNotStrip

/**
 * Routes pre-formatted lines into logcat so JS entries interleave with
 * native ones. Tag = the installed category.
 *
 * Marshalling only. The level map and the chunking that keeps logcat from
 * silently eating the tail of a long line live in [NativeConsoleWriter], which
 * imports nothing from Nitro and is therefore testable — the same split the
 * file sink uses, and for the same reason.
 */
@DoNotStrip
class HybridNativeConsoleSink : HybridNativeConsoleSinkSpec() {
  private val writer = NativeConsoleWriter()

  override fun install(subsystem: String, category: String) =
    writer.install(subsystem, category)

  override fun logBatch(levels: DoubleArray, messages: Array<String>) =
    writer.logBatch(levels, messages)
}

File: ios/HybridNativeConsoleSink.swift
import Foundation
import NitroModules

/// The Nitro end of the os_log sink: marshalling, and nothing else.
///
/// The level map and the chunking live in `NativeConsoleWriter`, which imports
/// nothing from Nitro and therefore runs under XCTest. Anything that grows a
/// decision in this file has been put in the wrong place — the same split the
/// file sink uses, for the same reason.
final class HybridNativeConsoleSink: HybridNativeConsoleSinkSpec {
  private let writer = NativeConsoleWriter()

  func install(subsystem: String, category: String) throws {
    writer.install(subsystem: subsystem, category: category)
  }

  func logBatch(levels: [Double], messages: [String]) throws {
    writer.logBatch(levels: levels, messages: messages)
  }
}
